lib/ws-channel.js

```diff
@@ -554,15 +586,17 @@ WebsocketChannel.prototype._fatalError = function(err) {
     err = new Error(err);
   debug('detach on fatal error: %s', err.message);
   this._detach();
+  this._closeWebsocket();
+  this._closed = err.message;
```

Comment:
I was thinking in particular of the `request()` callback up above, but it's debatable whether it would be useful anyway.